### PR TITLE
refactor: define maxℤ and minℤ manually

### PR DESCRIPTION
**Pull request by favonia**
# Description

Define maxℤ and minℤ manually by induction to facilitate reasoning about them.

## Checklist

Before submitting a merge request, please check the items below:

- [x] I've read [the contributing guidelines](https://github.com/plt-amy/1lab/blob/main/CONTRIBUTING.md).
- [x] The imports of new modules have been sorted with `support/sort-imports.hs` (or `nix run --experimental-features nix-command -f . sort-imports`).
- [x] All new code blocks have "agda" as their language.

### CODE REVIEW

**Comment by Lavenza** [comment]
[Pull request preview](https://preview.1lab.dev/333)

**Review by plt-amy — CHANGES_REQUESTED** [review]
The implementations look good, but could you move them to `Data.Int.Base`? And the proofs of the universal properties in `Data.Int.Properties`.

Now that they don't require any of the order-theory stuff, it'd be best if the dependency graph for "I just want `maxℤ`" could be untangled :slightly_smiling_face:

**Comment by favonia** [comment]
> And the proofs of the universal properties in `Data.Int.Properties`.

`Data.Int.Properties` would then depend on `Data.Int.Order` for the definition of the order. Should I put it in `Data.Int.Order.Properties`?

**Comment by plt-amy** [comment]
Oh, ugh. The proofs can stay where they are, I guess..

**Comment by favonia** [comment]
Is the current PR okay? I was thinking that things will be re-refactored once the lattice refactoring is done.

File: src/Data/Int.lagda.md [hunk 1]
```diff
@@ -12,5 +12,6 @@ module Data.Int where
 ```agda
 open import Data.Int.Properties public
 open import Data.Int.Order public
+open import Data.Int.Order.Properties public
 open import Data.Int.Base public
```
Comment (plt-amy) [inline src/Data/Int.lagda.md:16]:
```suggestion
open import Data.Int.Order.Properties public
open import Data.Int.Properties public
open import Data.Int.Order public
open import Data.Int.Base public
```

File: src/Data/Int/Order/Properties.lagda.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+```agda
+abstract
+```
+-->
+
+# Properties of the order of integers
+
+This module establishes further arithmetic and algebraic properties
+of the integers that involve the order. All proofs are straightforward
+case bashes; there are no further comments.
```
Comment (plt-amy) [inline src/Data/Int/Order/Properties.lagda.md:25]:
```suggestion
This module establishes further order-theoretic properties of the
[[integers]]. All proofs are straightforward case bashes; there are no
further comments.
```

**Review by plt-amy — CHANGES_REQUESTED** [review]

**Review by plt-amy — APPROVED** [review]

**Pull request merged**